src/mmds/src/lib.rs

```diff
@@ -137,6 +137,16 @@ pub fn parse_request(request_bytes: &[u8]) -> Response {
             }
         }
         Err(e) => match e {
+            RequestError::BodyWithoutPendingRequest => build_response(
```

Comment:
The changes in this file will need a small rebase and after that, we should test in `test_parse_request_bytes_error()` also the newly added error types.

Reply:
After rebasing and looking more carefully I realized that the overflow checks I'd added were superfluous, so I documented the safety of the operations instead